pootle/checks.py

```diff
@@ -0,0 +1,138 @@
+from django.core import checks
+
+
+# Minimum Translate Toolkit version required for Pootle to run.
+TTK_MINIMUM_REQUIRED_VERSION = (1, 11, 0)
+
+# Minimum Django version required for Pootle to run.
+DJANGO_MINIMUM_REQUIRED_VERSION = (1, 7, 4)
+
+# Minimum lxml version required for Pootle to run.
+LXML_MINIMUM_REQUIRED_VERSION = (2, 2, 2, 0)
+
+# Minimum Redis server version required.
+# Initially set to some minimums based on:
+# 1. Ubuntu 12.04LTS's version 2.8.4 (10.04LTS was too old for RQ)
+# 2. RQ requires >= 2.6.0, and
+# 3. Wanting to insist on at least the latest stable that devs are using i.e.
+#    2.8.* versions of Redis
+REDIS_MINIMUM_REQUIRED_VERSION = (2, 8, 4)
+
+
+@checks.register()
+def test_library_versions(app_configs, **kwargs):
+    from django import VERSION as django_version
+    from lxml.etree import LXML_VERSION as lxml_version
+    from translate.__version__ import ver as ttk_version
+
+    errors = []
+
+    if django_version < DJANGO_MINIMUM_REQUIRED_VERSION:
+        errors.append(checks.Critical("Your version of Django is too old.",
+            hint="Try pip install --upgrade Django",
+            id="pootle.E004",
+        ))
+
+    if lxml_version < LXML_MINIMUM_REQUIRED_VERSION:
+        errors.append(checks.Warning("Your version of lxml is too old.",
+            hint="Try pip install --upgrade lxml",
+            id="pootle.E005",
+        ))
+
+    if ttk_version < TTK_MINIMUM_REQUIRED_VERSION:
+        errors.append(checks.Critical("Your version of Translate Toolkit is too old.",
+            hint="Try pip install --upgrade translate-toolkit",
+            id="pootle.E006",
+        ))
+
+    return errors
+
+
+@checks.register()
+def test_redis(app_configs, **kwargs):
+    from django_rq.queues import get_queue
+    from django_rq.workers import Worker
+
+    errors = []
+
+    try:
+        queue = get_queue()
+        workers = Worker.all(queue.connection)
+    except Exception as e:
+        conn_settings = queue.connection.connection_pool.connection_kwargs
+        errors.append(checks.Critical("Could not connect to Redis (%s)" % (e),
+            hint="Make sure Redis is running on %(host)s:%(port)s" % (conn_settings),
+            id="pootle.E001",
+        ))
+    else:
+        if not workers or workers[0].stopped:
```

Comment:
I believe there needs to be a test to check if the current command being issued is `rqworker`. Otherwise, when running `rqworker` for the first time it'll display a redundant warning, which can be annoying, and more importantly alarming for some admins:
```python
$ ./manage.py rqworker
System check identified some issues:

WARNINGS:
?: (pootle.E002) No RQ Worker running.
	HINT: Run new workers with manage.py rqworker
```